submissions/team_api.py

```diff
@@ -0,0 +1,246 @@
+          'attempt_number' is the attempt number this submission represents for this question.
+          'submitted_at' represents the time this submission was submitted, which can be configured, versus the...
+          'created_at' date, which is when the submission is first created.
+          'submission_uuids' Is a list of the UUIDs of each of the individual `Submission` records that is created.
+
+    Raises:
+        TeamSubmissionRequestError: Raised when there are validation errors for the
+            student item or team submission. This can be caused by the student item
+            missing required values, the submission being too long, the
+            attempt_number is negative, or the given submitted_at time is invalid.
+        SubmissionRequestError: Raised when there are validation errors for the underlying
+            student item or submission. This can be caused by the same reason as
+            the TeamSubmissionRequestError
+        TeamSubmissionInternalError: Raised when submission access causes an
+            internal error.
+        TeamSubmissionInternalError: Raised when submission access causes an
+            internal error when creating the underlying submissions.
+
+    Examples:
+        >>>course_id = "course_1"
+        >>>item_id = "item_1"
+        >>>team_id = "A Team"
+        >>>submitting_user_id = "Tim"
+        >>>team_member_ids = ["Alice", "Bob", "Carol", "Tim"]
+        >>>answer = "The answer is 42."
+        >>> )
+        >>> create_submission_for_team(
+                course_id, item_id, team_id, submitting_user_id, team_member_ids, answer, datetime.utcnow, 1
+            )
+        {
+            'team_submission_uuid': 'blah',
+            'course_id': "course_1",
+            'item_id': "item_1",
+            'team_id': "A Team",
+            'submitted_by': "Tim",
+            'attempt_number': 1,
+            'submitted_at': datetime.datetime(2014, 1, 29, 17, 14, 52, 649284 tzinfo=<UTC>),
+            'created_at': datetime.datetime(2014, 1, 29, 17, 14, 52, 668850, tzinfo=<UTC>),
+            'answer': u'The answer is 42.',
+            'submission_uuids': ['alice-uuid', 'bob-uuid', 'carol-uuid', 'tim-uuid'],
+        }
+    """
+
+    # I have no clue what to do with attempt_number. The normal API checks if there are any duplicate submissions and
+    # incrememnts attempt_number, but we prevent duplicate team submissions, so I can't copy that logic flow.
+    # I thought maybe we should look for deleted submissions and incrememnt based of of that, but looking at
+    # production data I can't find a single case where that happens.
+    # For now ... I'm just gonna default it to 1?
+
+    model_kwargs = {
+        'course_id': course_id,
+        'item_id': item_id,
+        'team_id': team_id,
+        'submitted_by': submitting_user_id,
+        'attempt_number': attempt_number,
+    }
+    if submitted_at:
+        model_kwargs["submitted_at"] = submitted_at
+
+    try:
```

Comment:
When googling @transaction.atomic - I found the [doc page](https://docs.djangoproject.com/en/2.2/topics/db/transactions/) with the following quote (mid page) : 

> Avoid catching exceptions inside atomic!

Basically they recommend to double handle exceptions. Should we try and follow that pattern?

Reply:
I think that they're saying to follow that pattern when you're using the context manager inside of a try:catch. Either way, I tried it, and it doesn't seem like it really... fits?

plus, the example they give fits the following form: 
```
@transaction.atomic
def f():
   create_parent()
   try:
     with transaction.atomic():
        create_children()
    catch databasseerror:
       blah()
```

Reply:
we're more or less doing the same thing here, but we aren't trying to catch any eceptions when creating the children, so they'll jut propogate up. I think it should be okay?

Reply:
@nsprenkle do you have aany thoughts on this?

Reply:
Doing some more code scanning in our code base(platform) we have examples of what you are doing. I think we should be good to go.